src/main/java/org/sasanlabs/service/vulnerability/csrf/CSRFVulnerability.java

```diff
@@ -0,0 +1,218 @@
+                new GenericVulnerabilityResponseBean<String>(
+                        checkInputsToken(usrLiveToken, FIXED_TOKEN, usrId, usrPass, usrNewPass),
+                        true),
+                HttpStatus.OK);
+    }
+
+    @AttackVector(
+            vulnerabilityExposed = VulnerabilityType.CSRF,
+            description = "CSRF_VULNERABILITY_USER_INPUT1",
+            payload = "CSRF_PAYLOAD_LEVEL_5")
+    @VulnerableAppRequestMapping(
+            value = LevelConstants.LEVEL_5,
+            htmlTemplate = "LEVEL_5/CSRFVulnerability",
+            variant = Variant.SECURE)
+    public ResponseEntity<GenericVulnerabilityResponseBean<String>> getVulnerableCSRFLvl5(
+            @RequestParam(USR_LOGIN) String usrId,
+            @RequestParam(USR_PASS) String usrPass,
+            @RequestParam(USR_NEWPASS) String usrNewPass,
+            @RequestParam(RTK_PARAM) String usrLiveToken)
+            throws IOException {
+        if (usrLiveToken.equals("secretrequest")) {
+            scrt = Integer.toString(rand.nextInt(1000000000));
+            return new ResponseEntity<GenericVulnerabilityResponseBean<String>>(
+                    new GenericVulnerabilityResponseBean<String>(scrt, true), HttpStatus.OK);
+        } else {
+            return new ResponseEntity<GenericVulnerabilityResponseBean<String>>(
+                    new GenericVulnerabilityResponseBean<String>(
+                            checkInputsToken(usrLiveToken, scrt, usrId, usrPass, usrNewPass), true),
+                    HttpStatus.OK);
+        }
+    }
+
+    @AttackVector(
+            vulnerabilityExposed = VulnerabilityType.CSRF,
+            description = "CSRF_VULNERABILITY_USER_INPUT2",
+            payload = "CSRF_PAYLOAD_LEVEL_5")
+    @VulnerableAppRequestMapping(
+            value = LevelConstants.LEVEL_6,
+            htmlTemplate = "LEVEL_6/CSRFVulnerability",
+            variant = Variant.SECURE,
+            requestMethod = RequestMethod.POST)
+    public ResponseEntity<GenericVulnerabilityResponseBean<String>> getVulnerableCSRFLvl6(
+            @RequestParam(USR_LOGIN) String usrId,
+            @RequestParam(USR_PASS) String usrPass,
+            @RequestParam(USR_NEWPASS) String usrNewPass,
+            @RequestParam(RTK_PARAM) String usrLiveToken)
+            throws IOException {
+        if (usrLiveToken.equals("secretrequest")) {
+            scrt = Integer.toString(rand.nextInt(1000000000));
+            return new ResponseEntity<GenericVulnerabilityResponseBean<String>>(
+                    new GenericVulnerabilityResponseBean<String>(scrt, true), HttpStatus.OK);
+        } else {
+            return new ResponseEntity<GenericVulnerabilityResponseBean<String>>(
+                    new GenericVulnerabilityResponseBean<String>(
+                            checkInputsToken(usrLiveToken, scrt, usrId, usrPass, usrNewPass), true),
+                    HttpStatus.OK);
+        }
```

Comment:
I think you are doing 2 calls on clicking reset password, one for checking the token and the other for changing the password. I would suggest having one single call which does both things.